e2e/app/page/notebook-page.ts

```diff
@@ -226,17 +229,16 @@ export default class NotebookPage extends NotebookFrame {
     cellIndex: number,
     opts: { code?: string; codeFile?: string; timeOut?: number; markdownWorkaround?: boolean } = {}
   ): Promise<string> {
-    const cell = cellIndex === -1 ? await this.findLastCell() : this.findCell(cellIndex);
-    const inputCell = await cell.focus();
-
     const { code, codeFile, timeOut = 2 * 60 * 1000, markdownWorkaround = false } = opts;
-
-    let codeToRun;
-    if (code !== undefined) {
-      codeToRun = code;
-    } else if (codeFile !== undefined) {
-      codeToRun = fs.readFileSync(codeFile, 'ascii');
+    if (!!code && !!codeFile) {
```

Comment:
I think this error condition is incorrect. It's saying if code and codeFile are both provided, throw an error saying one of them is needed.

Also, double bang isn't necessary within conditional blocks since Javascript automatically coerces the type to boolean.

Reply:
Right. Function needs either `code` or `codeFile` to run, and not both defined.

Updated checks to:

```
if (code === undefined && codeFile === undefined) {
    throw new Error('Code or codeFile parameter is required in runCodeCell method.');
}
if (code !== undefined && codeFile !== undefined) {
    throw new Error('Code and codeFile parameters are both defined. Define one of two in runCodeCell method.');
}
```